tests/cypress/e2e/gui/testsGui.cy.js

```diff
@@ -0,0 +1,114 @@
+describe('Validate the GUI', () => {
+  beforeEach(() => {
+    cy.intercept({
+      method: 'GET',
+      url: 'http://localhost:3001/customers?**',
+    }).as('getCustomers');
+    cy.openEngageSphere()
+    cy.visit('/')
+  })
+
+  it('Keep filters when returning from the customer details view', () => {
+    // Filter the companies
+    cy.clock()
+    cy.get('[data-testid="size-filter"]').select('Medium').should('have.value', 'Medium')
```

Comment:
```suggestion
    cy.get('[data-testid="size-filter"]').select('Medium')
```

Tal verificação é desnecessária.

O comportamento padrão de um campo do tipo `select` ao selecionar uma `option` é que tal `option` seja selecionada.